cppcache/src/CacheXmlParser.cpp

```diff
@@ -494,8 +491,7 @@ void CacheXmlParser::setAttributes(Cache *) {}
  *
  */
 void CacheXmlParser::create(Cache *cache) {
-  // use DeleteObject class to delete cacheCreation_ in case of exceptions
-  DeleteObject<CacheXmlCreation> delCacheCreation(cacheCreation_);
+  std::unique_ptr<CacheXmlCreation> delCacheCreation(cacheCreation_);
```

Comment:
Yes, It looks like the purpose here was to prevent the `cache` from being leaked if there was an exception. I suspect the caller is already doing this if it's a smart pointer already. Perhaps a `Cache&` is even more appropriate to void even needing the `nullptr` check.